tests/beaker_tests/cisco_interface_ospf/ospfintf_provider_defaults.rb

```diff
@@ -62,21 +62,18 @@
 result = 'PASS'
 testheader = 'OSPFINTF Resource :: All Attributes Defaults'
 
+if ethernet_info_get.nil? || ethernet_info_get[:slot].nil?
+  msg = 'Unable to find suitable ethernet module required for this test.'
+  prereq_skip(testheader, self, msg)
+end
+interface = "ethernet#{ethernet_info_get[:slot]}/1"
```

Comment:
I like this but I'd prefer that you add another helper method to reduce handling in the calling methods; ie. the callers should just ask for an interface and the helper can do the skip check. This is what I have in mind:

```
# Find an appropriate test interface. Optionally raise a skip.
def find_interface(type='ethernet', skipcheck=true)
  case type
  when /ethernet/i
    intf = ethernet_info_get
    intf = "ethernet[intf[:slot]}/1" unless intf.nil?
  end
  if skipcheck && intf.nil?
    msg = 'Unable to find suitable interface module for this test.'
    prereq_skip(testheader, self, msg)
  end
end
```

Now the callers just need to do this:

```
interface = find_interface
```